core/src/main/java/org/elasticsearch/discovery/zen/fd/NodesFaultDetection.java

```diff
@@ -256,14 +260,15 @@ public String executor() {
         public void messageReceived(PingRequest request, TransportChannel channel) throws Exception {
             // if we are not the node we are supposed to be pinged, send an exception
             // this can happen when a kill -9 is sent, and another node is started using the same port
-            if (!localNode.getId().equals(request.nodeId)) {
-                throw new IllegalStateException("Got pinged as node [" + request.nodeId + "], but I am node [" + localNode.getId() + "]");
+            if (!localNode.equals(request.node())) {
+                throw new IllegalStateException("Got pinged as node " + request.node() + "], but I am node " + localNode );
```

Comment:
I like that these exceptions are going to be much more useful with all node information contained (no more cryptic ids).